src/backend/src/filesystem/hl_operations/hl_mkdir.js

```diff
@@ -275,11 +275,20 @@ class HLMkdir extends HLFilesystemOperation {
         console.log('USING PARENT', parent_node.selector.describe());
         let target_basename = _path.basename(values.path);
 
+        // "top_parent" is the immediate parent of the target directory
+        // (e.g: /home/foo/bar -> /home/foo)
         const top_parent = values.create_missing_parents
             ? await this._create_top_parent({ top_parent: parent_node })
             : await this._get_existing_top_parent({ top_parent: parent_node })
             ;
 
+        if ( top_parent.isRoot ) {
```

Comment:
```suggestion
        // TODO: this can be removed upon completion of: https://github.com/HeyPuter/puter/issues/1352
        if ( top_parent.isRoot ) {
```